src/index.js

```diff
@@ -25,144 +25,137 @@ import InstructorsOnly from './components/InstructorsOnly'
 import LoggedOut from './components/LoggedOut'
 import RouteNotFound from './components/RouteNotFound'
 
-export default class Routes extends Component {
+const App = () => {
 
-  state = {
-    user: false,
-  }
-
-  componentDidMount() {
-    login(this.addUser)
-  }
+  const userUrl = login()
 
-  addUser = (user) => {
-    this.setState({user})
-    if (process.env.NODE_ENV === 'production') {
-      initializeErrorTracking(user.instructor_id)
-    }
+  if(!userUrl) {
+    return <LoggedOut />
   }
 
-  render() {
-
-    const {user} = this.state
-
-    if(!user) {
-      return <LoggedOut />
-    }
-
-    if(user && !user.instructor_id) {
-      return <InstructorsOnly />
-    }
-
-    return (
-      <BrowserRouter>
-        <div>
-
-          <Navigation
-            items={[
-              {
-                text: overviewTitleText,
-                action: '/',
-              },
-              {
-                text: newLessonsTitleText,
-                action: '/lessons/new',
-              },
-              {
-                text: instructorsTitleText,
-                action: '/instructors',
-              },
-              {
-                text: guideTitleText,
-                action: guideUrl,
-              },
-              {
-                text: chatTitleText,
-                action: chatUrl,
-              },
-              {
-                text: logOutTitleText,
-                action: logout,
-              },
-            ]}
-          />
-
-          <Main>
-
-            <Switch>
-
-              <Route 
-                exact
-                path='/'
-                render={() => (
-                  <Request url={`/api/v1/instructors/${user.instructor_id}`}>
-                    {({data}) => (
-                      <div>
-                        <Overview instructor={data} />
-                      </div>
-                    )}
-                  </Request>
-                )}
+  return (
+    <Request url={userUrl}>
+      {({data}) => {
```

Comment:
`data` is a variable shadow here, as routes also have <Request /> and data. It works but would be good if there was a simple way to make it more clear; wish we could rename while destructuring (`{data as user}`). Might make a `dataName` prop to pass to requests so returned render callback param has a more specific name than just `data` depending on instance.